src/main/java/core/basesyntax/service/impl/DataValidatorService.java

```diff
@@ -0,0 +1,47 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.exception.InvalidDataFormatException;
+import core.basesyntax.exception.InvalidOperationException;
+import java.util.List;
+
+public class DataValidatorService {
```

Comment:
your validator breaks the single responsibility principle because it validates a lot of different data. Add all checks where you need them in your code and delete this class.
Also, it decreases the readability of your code. Here is an example: you use the method `isValidQuantityToPurchase`, but to understand what this method does - I need to go to another file. Also, you create a new temporary variable every time for purchase operation. Just imagine if you need to process 2 billion purchase operations. what happens with your memory?